src/viking.js

```diff
@@ -1,11 +1,119 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health
+    this.strength = strength
+}
+
+Soldier.prototype.attack = function () {
+    return this.strength
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health -= damage
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
```

Comment:
There is one line missing to take the methods only! If not now the constructor of the Viking is the Soldier not the Viking!